### PR TITLE
refactor(shorebird_cli): Add `IosFrameworkPatcher`

### PR DESCRIPTION
**Pull request by bryanoltman**
## Description

Adds IosFrameworkPatcher to support patching hybrid iOS with the new patch command.

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

**Comment by bryanoltman** [comment]
Closing and reopening to …maybe? fix CI

File: packages/shorebird_cli/test/src/commands/patch_new/patch_new_command_test.dart [hunk 1]
```diff
@@ -136,6 +140,8 @@ void main() {
         () => artifactManager.downloadFile(any()),
       ).thenAnswer((_) async => File(''));
 
+      when(() => cache.updateAll()).thenAnswer((_) async => {});
```
Comment (felangel) [inline packages/shorebird_cli/test/src/commands/patch_new/patch_new_command_test.dart:143]: should we also verify that `cache.updateAll` is called later in a test?

**Review by felangel — APPROVED** [review]
LGTM 👍

File: packages/shorebird_cli/test/src/commands/patch_new/patch_new_command_test.dart (hunk 1 above)
Comment (bryanoltman) [reply packages/shorebird_cli/test/src/commands/patch_new/patch_new_command_test.dart:143]: Good call, done

**Review by felangel — APPROVED** [review]

File: packages/shorebird_cli/test/src/commands/patch_new/patch_new_command_test.dart [hunk 2]
```diff
@@ -419,6 +427,19 @@ void main() {
             () => patcher.buildPatchArtifact(),
           ]);
         });
+
+        test('updates cache with both default and release Flutter revisions',
```
Comment (felangel) [inline packages/shorebird_cli/test/src/commands/patch_new/patch_new_command_test.dart:431]: Do we need to do the same thing for release new?

Comment (bryanoltman) [reply packages/shorebird_cli/test/src/commands/patch_new/patch_new_command_test.dart:431]: Probably. Adding.

Comment (bryanoltman) [reply packages/shorebird_cli/test/src/commands/patch_new/patch_new_command_test.dart:431]: Done.

**Review by erickzanardo — APPROVED** [review]

**Pull request merged**